webapp/src/components/sidebar_right/github_items.tsx

```diff
@@ -1,47 +1,120 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React from 'react';
-import PropTypes from 'prop-types';
+import * as React from 'react';
+
+import * as CSS from 'csstype';
 
 import {Badge, Tooltip, OverlayTrigger} from 'react-bootstrap';
+import {Theme} from 'mattermost-redux/types/preferences';
 import {makeStyleFromTheme, changeOpacity} from 'mattermost-redux/utils/theme_utils';
-import {GitPullRequestIcon, IssueOpenedIcon} from '@primer/octicons-react';
+import {GitPullRequestIcon, IssueOpenedIcon, IconProps} from '@primer/octicons-react';
 
-import {formatTimeSince} from 'utils/date_utils';
+import {formatTimeSince} from '../../utils/date_utils';
 
-import CrossIcon from 'images/icons/cross.jsx';
-import DotIcon from 'images/icons/dot.jsx';
-import TickIcon from 'images/icons/tick.jsx';
-import SignIcon from 'images/icons/sign.jsx';
-import ChangesRequestedIcon from 'images/icons/changes_requested.jsx';
+import CrossIcon from '../../images/icons/cross';
+import DotIcon from '../../images/icons/dot';
+import TickIcon from '../../images/icons/tick';
+import SignIcon from '../../images/icons/sign';
+import ChangesRequestedIcon from '../../images/icons/changes_requested';
 import {getLabelFontColor} from '../../utils/styles';
 
-function GithubItems(props) {
+const notificationReasons = {
+    assign:	'You were assigned to the issue',
+    author:	'You created the thread.',
+    comment:	'You commented on the thread.',
+    invitation:	'You accepted an invitation to contribute to the repository.',
+    manual:	'You subscribed to the thread (via an issue or pull request).',
+    mention:	'You were specifically @mentioned in the content.',
+    review_requested:	'You were requested to review a pull request.',
+    security_alert: 'GitHub discovered a security vulnerability in your repository.',
+    state_change: 'You changed the thread state.',
+    subscribed:	'You are watching the repository.',
+    team_mention:	'You were on a team that was mentioned.',
+};
+
+interface Label {
+    id: number;
+    name: string;
+    color: CSS.Properties;
+}
+
+interface User {
+    login: string;
+}
+
+interface Review {
+    state: string;
+    user: User;
+}
+
+interface Item {
+    url: string;
+    number: number;
+
+    id: number;
+    title: string;
+    created_at: string;
+    html_url: string;
+    repository_url?: string;
+    user: User;
+    owner?: User;
+    milestone?: {
+        title: string;
+    }
+    repository?: {
+        full_name: string;
+    }
+    labels?: Label[];
+
+    // PRs
+    status?: string;
+    mergeable?: boolean;
+    requestedReviewers?: string[];
+    reviews?: Review[];
+
+    // Assignments
+    pullRequest?: undefined;
```

Comment:
I think `unknown` would be a better type for this

```suggestion
    pullRequest?: unknown;
```